test/suite/test_import10.py

```diff
@@ -0,0 +1,89 @@
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import wiredtiger
+from wtscenario import make_scenarios
+from test_import01 import test_import_base
+
+# test_import10.py
+#    Run import/export while backup cursor is open.
+class test_import10(test_import_base):
+    uri = 'test_import10'
+    create_config = 'allocation_size=512,key_format=i,value_format=i'
+    scenarios = make_scenarios([
+        ('import_with_metadata', dict(repair=False)),
+        ('import_repair', dict(repair=True)),
+    ])
+
+    def test_import_with_open_backup_cursor(self):
+        # Create and populate the table.
+        table_uri = 'table:' + self.uri
+        self.session.create(table_uri, self.create_config)
+        cursor = self.session.open_cursor(table_uri)
+        for i in range(1, 1000):
+            cursor[i] = i
+        cursor.close()
+        self.session.checkpoint()
+
+        # Export the metadata for the file.
+        file_uri = 'file:' + self.uri + '.wt'
+        c = self.session.open_cursor('metadata:', None, None)
+        original_db_table_config = c[table_uri]
+        original_db_file_config = c[file_uri]
+        c.close()
+
+        self.session.drop(table_uri, 'remove_files=false')
+
+        # Verify the table no longer exists in the database by attempting to open a cursor.
+        self.assertRaisesException(wiredtiger.WiredTigerError,
+            lambda: self.session.open_cursor(table_uri))
+
+        # Open backup cursor.
+        bkup_c = self.session.open_cursor('backup:', None, None)
+
+        # First construct the config string for the default or repair import scenario,
+        # then call create to import the table.
+        if self.repair:
+            import_config = 'import=(enabled,repair=true)'
+        else:
+            import_config = '{},import=(enabled,repair=false,file_metadata=({}))'.format(
+                original_db_table_config, original_db_file_config)
+        self.session.create(table_uri, import_config)
+
+        # Verify object.
+        self.session.verify(table_uri)
+
+        # Check that the data got imported correctly.
+        cursor = self.session.open_cursor(table_uri)
+        for i in range(1, 1000):
+            self.assertEqual(cursor[i], i)
+        cursor.close()
+        bkup_c.close()
```

Comment:
While this meets the minimum of the test description, it might be useful to walk the backup cursor sometime after the import-create and make sure that the file/table is not one of the returned files for backup.

Reply:
done!